src/fragments/lib/storage/flutter/list.mdx

```diff
@@ -34,10 +12,12 @@ Future<void> listAlbum() async {
   try {
```

Comment:
This example doesn't make much sense. If you call this multiple times such that you reach the end of the list, it will continue listing the last page, no?

Reply:
Agreed, this example may not behave exactly as what the leading paragraph described. Could be something like

```dart
import 'package:amplify_flutter/amplify_flutter.dart';
import 'package:amplify_storage_s3/amplify_storage_s3.dart';

Future<void> listAlbum() async {
  try {
    late StorageListResult result;

    result = await Amplify.Storage.list(
      path: 'album/',
      options: const StorageListOptions(
        accessLevel: StorageAccessLevel.private,
        pageSize: 50,
        pluginOptions: S3ListPluginOptions(
          excludeSubPaths: true,
        ),
      ),
    ).result;

    safePrint('Listed items: ${result.items}');

    while (result.hasNextPage) {
      result = await Amplify.Storage.list(
        path: 'album/',
        options: StorageListOptions(
          accessLevel: StorageAccessLevel.private,
          pageSize: 50,
          nextToken: result.nextToken,
          pluginOptions: const S3ListPluginOptions(
            excludeSubPaths: true,
          ),
        ),
      ).result;

      safePrint('Listed items: ${result.items}');
    }
  } on StorageException catch (e) {
    safePrint('Error listing files: ${e.message}');
    rethrow;
  }
}
```

Or replace with a different example, how about the `listAlbum()` returns a function, when call it prints the next page?

```dart
Future<void Function()?> listAlbum([String? nextToken]) async {
  try {
    final result = await Amplify.Storage.list(
      path: 'album/',
      options: StorageListOptions(
        accessLevel: StorageAccessLevel.private,
        pageSize: 50,
        nextToken: nextToken,
        pluginOptions: const S3ListPluginOptions(
          excludeSubPaths: true,
        ),
      ),
    ).result;

    safePrint('Listed items: ${result.items}');

    if (result.hasNextPage) {
      return () {
        listAlbum(result.nextToken);
      };
    }
  } on StorageException catch (e) {
    safePrint('Error listing files: ${e.message}');
    rethrow;
  }
  return null;
}

final listNextPage = await listAlbum();
if (listNextPage != null) {
  listNextPage();
}
```